safe_transaction_service/history/indexers/events_indexer.py

```diff
@@ -22,9 +24,11 @@ class EventsIndexer(EthereumIndexer):
     Indexes Ethereum events
     """
 
-    IGNORE_ADDRESSES_ON_LOG_FILTER: bool = (
-        False  # If True, don't use addresses to filter logs
-    )
+    # If True, don't use addresses to filter logs
+    # Be careful, some nodes have limitations
+    # https://docs.nodereal.io/nodereal/meganode/api-docs/bnb-smart-chain-api/eth_getlogs-bsc
+    # https://docs.infura.io/infura/networks/ethereum/json-rpc-methods/eth_getlogs#limitations
+    IGNORE_ADDRESSES_ON_LOG_FILTER: bool = False
```

Comment:
Well... It's more than a configuration parameter and that's why I decided to use it there instead of reading it from environment variables. `EventsIndexer` is a class meant to be updated (it's an abstract class) and `IGNORE_ADDRESSES_ON_LOG_FILTER` depends more on the class implementing it. For example:
- For `SafeEventsIndexer`, we only search for the events (only using the main `topic`) and we don't care about the addresses: https://web3py.readthedocs.io/en/latest/filters.html#event-log-filters
- For `Erc20EventsIndexer` we cannot do that, as there are a lot of events and we reach the node limit (some nodes limit to `10000` or `50000`, so we need to search using the addresses even if that's slower. We might need to do the same for `SafeEventsIndexer` at some point.

I'm still not quite happy about this environment variable and I totally get your point. WDYT?

Reply:
I see, thanks for the detailed and clear explanation!

My two cents: could it be an option to avoid exposing (or even [mutating](https://github.com/safe-global/safe-transaction-service/pull/966/files#diff-446430fbcddd527c120913a894ccba97ef719c5dc05b4ce9e48f68500e72df1bR366)) the state of the concrete indexer? I mean, instead of reading `IGNORE_ADDRESSES_ON_LOG_FILTER` from outside the class, just ask the indexer for the addresses directly by implementing a method which returns them? That way we could favor encapsulation on the concrete classes (indexers), and hiding its inner logic from the caller (IndexService).

If I’m getting this right, that way we could even get rid of the boolean and let each indexer its own logic to obtain the addresses. Maybe I’m missing something here, I don't have a lot of expertise on how Python hierarchy works.

Reply:
Yes, agree on not mutating. I don't like it either at all. The problem is that you need to provide the addresses for all the flow (reading them from database, checking blocks, updating them when it's over) so the only step that changes is the last one. Maybe I'm not getting you, would you like to do a PR after I merge this?

Reply:
Sure, I can try it. Maybe I'm not getting it right but at least I can try to refactorize this and get familiar with the codebase :)

Reply:
Awesome :)